src/device/usbd.c

```diff
@@ -182,7 +182,31 @@ static usbd_class_driver_t const _usbd_driver[] =
     .reset            = dfu_rtd_reset,
     .open             = dfu_rtd_open,
     .control_xfer_cb  = dfu_rtd_control_xfer_cb,
-    .xfer_cb          = dfu_rtd_xfer_cb,
+    .xfer_cb          = NULL,
+    .sof              = NULL
+  },
+  #endif
+
+  #if CFG_TUD_DFU_MODE
+  {
+    DRIVER_NAME("DFU-MODE")
+    .init             = dfu_mode_init,
+    .reset            = dfu_mode_reset,
+    .open             = dfu_mode_open,
+    .control_xfer_cb  = dfu_mode_control_xfer_cb,
+    .xfer_cb          = NULL,
+    .sof              = NULL
+  },
+  #endif
+
+  #if CFG_TUD_DFU_RUNTIME_AND_MODE
```

Comment:
hmm, what is the scenario you have in mind for this `RUNTIME_AND_MODE`. Can you elaborate it more, I thought it is either dfu runtime or dfu mode (bootloader)

Reply:
You are correct in that dfu runtime and dfu mode are exclusive.  After I split the RT and Mode, I ran into an issue where because the tusb_config.h had:
```
#define CFG_TUD_DFU_RUNTIME 1
#define CFG_TUD_DFU_MODE    1
```
Which resulted in both DFU classes being initialized when `tud_task()` was called.

One possible solution to this would be to have separate builds, one for the app with RT and another for the bootloader with Mode.  In the event than an application does not want to have 2 separate builds, I added this this `RUNTIME_AND_MODE` "class" which allows for the code for both to exist in a single build, but only 1 class is initialized at runtime.  When this DFU class is initialized, it will ask the app if it should run as RT or Mode.  After this check, the class will behave only as RT or Mode.

The name can be changed to prevent confusion.  CFG_TUD_DFU_DYNAMIC or something.

Reply:
I don't think we would need to ask the app, I wonder why would bptj CFG_TUD_DFU_RUNTIME and CFG_TUD_DFU_MODE are enabled at the same time. And what would be the issue with both dfu mode and dfu rt are initialized. It really doesn't matter, the usb configuration descriptor will dictate which interface e.g itf = 0 for the DFU mode and which is used for DFU runtime should they are both enabled.

Please remove the the `RUNTIME_AND_MODE` we will treat each one as separated driver like msc vs cdc.

Reply:
Resolved in 7b45b38